docker/test/performance-comparison/compare.sh

```diff
@@ -284,13 +284,21 @@ function run_tests
         # Use awk because bash doesn't support floating point arithmetic.
         profile_seconds=$(awk "BEGIN { print ($profile_seconds_left > 0 ? 10 : 0) }")
 
+        if [ "$(grep -c $(basename $test) changed-test-definitions.txt)" -gt 0 ]
+        then
+          # Run all queries from changed test files to ensure that all new queries will be tested.
```

Comment:
Hm, but this will not only run "all new queries" but "all queries" if there are changes in perf tests, since perf tests will run only specific tests if and only if, there were no changes except for perf tests:

https://github.com/ClickHouse/ClickHouse/blob/9f1ee52b0095c462604ff1be892b6a55916f7d14/docker/test/performance-comparison/compare.sh#L199

So:
- `tests/performance/avg.xml` changed - run only specific test, w/o `--max_queries` limit
- `tests/performance/avg.xml` and `src/*` - **run all testes and w/o `--max_queries` lmitt**

And usually there are PRs with new tests, not just changes in existing tests, and so they will takes significant amount of time.

So either condition above should check for `"$(wc -l < other-changed-files.txt)" -eq 0`, or, and this looks better to me, tests with changes should be runned w/o query limit, others can be run with limit.

Reply:
> tests with changes should be runned w/o query limit, others can be run with limit

that was our intention. this `if` determines the limit for each query individually, and for untouched perf tests limit should be applied.

Reply:
Oh, I see, thanks! (misread the patch due to small context).